src/traces/main.cpp

```diff
@@ -0,0 +1,164 @@
+#include <iostream>
+#include <random>
+#include <vector>
+
+#include <time.h>
+
+#include "../common/logger.hpp"
+#include "../common/status.hpp"
+#include "../common/types.hpp"
+
+#include "../sequential/sequential.hpp"
+
+#include "../parallel/parallel.hpp"
+
+#include <iostream>
+#include <optional> // std::optional i think is c++17
+#include <pthread.h>
+
+// structure to pass data to threads
+struct OperationData {
+    enum OperationType { INSERT, SEARCH, DELETE };
+
+    OperationType operation;
+    KeyType key;
+    ValueType value;
+};
+
+// Define the structure to pass data to threads
+struct ThreadData {
+    ParallelRobinHoodHashTable *hashTable;
+    OperationData *operations;
+    size_t numOperations;
+
+    int threadID;
+};
+
+
+// Function to convert OperationType to string for better output
+const char* operationToString(OperationData::OperationType op) {
+    switch (op) {
+        case OperationData::INSERT:
+            return "INSERT";
+        case OperationData::SEARCH:
+            return "SEARCH";
+        case OperationData::DELETE:
+            return "DELETE";
+    }
+    return "UNKNOWN";
+}
```

Comment:
Maybe there's a function to directly print them? E.g. https://cplusplus.com/forum/general/263696/